elide-core/src/main/java/com/yahoo/elide/core/LifecycleHookInvoker.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2018, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.core;
+
+import com.yahoo.elide.functions.LifeCycleHook;
+import io.reactivex.Observer;
+import io.reactivex.disposables.Disposable;
+
+import java.lang.annotation.Annotation;
+import java.util.Collection;
+
+/**
+ * RX Java Observer which invokes a lifecycle hook function.
+ */
+public class LifecycleHookInvoker implements Observer<CRUDEvent> {
+
+    private EntityDictionary dictionary;
+    private Class<? extends Annotation> annotation;
+
+    public LifecycleHookInvoker(EntityDictionary dictionary, Class<? extends Annotation> annotation) {
+        this.dictionary = dictionary;
+        this.annotation = annotation;
+    }
+
+    @Override
+    public void onSubscribe(Disposable disposable) {
+        //NOOP
+    }
+
+    @Override
+    public void onNext(CRUDEvent event) {
+
+        //Collect all the hooks that are keyed on a specific field.
+        Collection<LifeCycleHook> hooks = dictionary.getTriggers(
+                event.getResource().getResourceClass(),
+                this.annotation,
+                event.getFieldName());
+
+        //Collect all the hooks that are keyed on any field.
+        if (!event.getFieldName().isEmpty()) {
+            hooks.addAll(dictionary.getTriggers(event.getResource().getResourceClass(), this.annotation));
```

Comment:
is it safe to modify the returned collection like this? if it changes to be immutable or changes from list to set, for instance this would alter the semantics of the `addAll` logic below. Should we create a new one with this?